pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1609,15 +1681,35 @@ func (gce *GCECloud) DiskIsAttached(diskName, instanceID string) (bool, error) {
 	return false, nil
 }
 
-func (gce *GCECloud) getDisk(diskName string) (*compute.Disk, error) {
-	return gce.service.Disks.Get(gce.projectID, gce.zone, diskName).Do()
+func (gce *GCECloud) resolveDisk(diskName string) (*gceDisk, error) {
```

Comment:
And as per previous comments, I think that we can always easily pass in a zone here too, and get around the problem that you mention.